src/components/program/tests/ProgramEndorsements.test.jsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { screen, render, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { UserSubsidyContext } from '../../enterprise-user-subsidy';
+import { ProgramContextProvider } from '../ProgramContextProvider';
+import ProgramEndorsements from '../ProgramEndorsements';
+
+jest.mock('react-router-dom', () => ({
+  useLocation: jest.fn(),
+}));
+
+/* eslint-disable react/prop-types */
+const ProgramEndorsementsWithContext = ({
+  initialAppState = {},
+  initialProgramState = {},
+  initialUserSubsidyState = {},
+}) => (
+  <AppContext.Provider value={initialAppState}>
+    <UserSubsidyContext.Provider value={initialUserSubsidyState}>
+      <ProgramContextProvider initialState={initialProgramState}>
+        <ProgramEndorsements />
+      </ProgramContextProvider>
+    </UserSubsidyContext.Provider>
+  </AppContext.Provider>
+);
+/* eslint-enable react/prop-types */
+
+describe('<ProgramEndorsements />', () => {
+  const initialAppState = {
+    enterpriseConfig: {
+      slug: 'test-enterprise-slug',
+    },
+  };
+  const initialProgramState = {
```

Comment:
are we using all the fields/attributes in the component? Can we have only those field/attributes which are being used in the component and remove others?